php/commands/role.php

```diff
@@ -367,6 +383,167 @@ private static function persistence_check() {
 		if ( !$wp_roles->use_db )
 			WP_CLI::error( "Role definitions are not persistent." );
 	}
+
+	private static function get_role_capability( $role_key ) {
```

Comment:
For comparing the default capability and exist capability of a role in the database, we needed list/array of default capability of a role for dry-run. As for reset role we first remove role and store the capability in the array, then we add the roles with default capability again. After that, we check count using that array.
Here I didn't found any WordPress core function to fetch the default capability of the role. We can add this function in core.
There is one another way without using this method, we can add the removed capability after resetting role for dry-run. But I was not sure if that's a good idea.